src/main.js

```diff
@@ -41,42 +45,59 @@ document.addEventListener('DOMContentLoaded', () => {
   handleLocation();
 });
 
-export const getUrl = (event) => {
-  event.preventDefault();
-  const url = new URL(event.currentTarget.href);
+export const getUrl = (stringUrl) => {
+  const url = new URL(stringUrl);
+
   url.pathname = `${basePath}${url.pathname}`;
   return url;
 };
 
-export const setParams = (event, url) => {
-  const dataset = event.currentTarget.dataset;
+export const setParams = (queries = {}, url) => {
   const searchKeys = Object.values(SEARCH_PARAMS);
-  Object.keys(dataset).forEach((key) => {
+  Object.keys(queries).forEach((key) => {
     if (searchKeys.includes(key)) {
-      url.searchParams.set(key, dataset[key]);
+      url.searchParams.set(key, queries[key]);
     }
   });
 };
 
 export const handleLocation = async () => {
-  const { pathname, search } = window.location;
+  const { search } = window.location;
   const urlParams = new URLSearchParams(search);
-  if (urlParams.has('name')) {
+  if (!urlParams.has('filter')) {
+    const searchEl = document.querySelector('.search-exercises');
+    if (!searchEl) appendSearch(urlParams);
     getExercises(urlParams);
     return;
   }
   getFilters(urlParams);
   getFavorites(urlParams);
+  removeSearch();
 };
 
 export const pushState = (url) => {
   window.history.pushState({}, '', url);
   handleLocation();
 };
 
-export const handleClick = (event) => {
-  const url = getUrl(event);
-  setParams(event, url);
+export const handleClickFilter = (event) => {
+  event.preventDefault();
+  const url = getUrl(event.currentTarget.href);
+  setParams(event.currentTarget.dataset, url);
+  pushState(url);
+};
+
+export const handleClickCategory = (event) => {
+  event.preventDefault();
+  const url = getUrl(window.location.origin);
+  setParams(
```

Comment:
this function already sets params, handleClickCategory should be located in categores page or home

Reply:
fixed